Libraries/Azure.psm1

```diff
@@ -2849,223 +2849,225 @@ function Add-AzureAccountFromSecretsFile {
-
-		foreach ( $vmData in $allVMData ) {
-			$ResourceGroupName = $vmData.ResourceGroupName
-			$RoleName = $vmData.RoleName
-			$RoleSize = $vmData.InstanceSize
-
-			#Copy and run test file
-			$null = Copy-RemoteFiles -upload -uploadTo $vmData.PublicIP -port $vmData.SSHPort -files ".\Testscripts\Linux\CollectLogFile.sh" -username $user -password $password
-			$null = Run-LinuxCmd -username $user -password $password -ip $vmData.PublicIP -port $vmData.SSHPort -command "bash CollectLogFile.sh" -ignoreLinuxExitCode
-
-			#download the log files
-			$null = Copy-RemoteFiles -downloadFrom $vmData.PublicIP -port $vmData.SSHPort -username $user -password $password -files "$($vmData.RoleName)-*.txt" -downloadTo "$LogDir" -download
-
-			# Upload files in data subfolder to Azure.
-			$destfolder = "bootPerf"
-			$containerName = "logs"
-			$blobContext = New-AzStorageContext -StorageAccountName $storageAccountName -StorageAccountKey $storageAccountKey
-
-			$ticks = (Get-Date).Ticks
-			$fileName = "$LogDir\$($vmData.RoleName)-waagent.log.txt"
-			$blobName = "$destfolder/$($fileName.Replace("waagent","waagent-$ticks") | Split-Path -Leaf)"
-			$null = Set-AzStorageBlobContent -File $filename -Container $containerName -Blob $blobName -Context $blobContext -Force
-			$WALAlogFile = "https://$storageAccountName.blob.core.windows.net/$containerName/$destfolder/$($fileName.Replace("waagent","waagent-$ticks") | Split-Path -Leaf)"
-			Write-LogInfo "Upload file to Azure: Success: $WALAlogFile"
-			$fileName = "$LogDir\$($vmData.RoleName)-dmesg.txt"
-			$blobName = "$destfolder/$($fileName.Replace("dmesg","dmesg-$ticks") | Split-Path -Leaf)"
-			$null = Set-AzStorageBlobContent -File $filename -Container $containerName -Blob $blobName -Context $blobContext -Force
-			$kernelLogFile = "https://$storageAccountName.blob.core.windows.net/$containerName/$destfolder/$($fileName.Replace("dmesg","dmesg-$ticks") | Split-Path -Leaf)"
-			Write-LogInfo "Upload file to Azure: Success: $kernelLogFile"
-
-
-			# Analyse
-			$waagentFile = "$LogDir\$($vmData.RoleName)-waagent.log.txt"
-			$waagentLogs = Get-Content -Path $waagentFile
-
-			#region Detect the WALA identifiers
-			foreach ($line in $waagentLogs.Split("`n")) {
-				foreach ( $keyword in $TextIdentifiers.identifiers.waagent.ProvisionStarted.keyword ) {
-					if ($line -imatch $keyword) {
-						$walaStartIdentifier = $keyword
+	# Get the Database data
+	$dataSource = $Global:XMLSecrets.secrets.DatabaseServer
+	$dbuser = $Global:XMLSecrets.secrets.DatabaseUser
+	$dbpassword = $Global:XMLSecrets.secrets.DatabasePassword
+	$database = $Global:XMLSecrets.secrets.DatabaseName
+	if ($dataSource -and $dbuser -and $dbpassword -and $database) {
```

Comment:
Add pre-check for  '$dataSource -and $dbuser -and $dbpassword -and $database',  is just want to speed up the deployment, when LISAv2 user does not provide Database information in secret file.